File: array-string/remove-duplicates-from-sorted-array-2/main.py
from typing import List


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        k = 0
        dup = 1
        for i in range(1, len(nums)):
            if nums[i] != nums[k]:
                k += 1
                nums[k] = nums[i]
                dup = 1
            elif dup < 2:
                k += 1
                nums[k] = nums[i]
                dup += 1
        return k + 1

File: array-string/remove-duplicates-from-sorted-array-2/main.go
package main

func removeDuplicates(nums []int) int {
	n := len(nums)
	k := 0
	dup := 1
	for i := 1; i < n; i++ {
		if nums[i] != nums[k] {
			k++
			nums[k] = nums[i]
			dup = 1
		} else if dup < 2 {
			k++
			nums[k] = nums[i]
			dup++
		}
	}
	return k
}
